example-crates/example_cli/src/lib.rs

```diff
@@ -183,18 +183,17 @@ impl core::fmt::Display for Keychain {
     }
 }
 
-#[allow(clippy::type_complexity)]
+// Alias changeset info in `create_tx`
+pub type ChangeSet = (keychain::ChangeSet<Keychain>, (Keychain, u32));
```

Comment:
Yes, I like that. However, since three of these fields are either all `Some` or all `None`, it might make sense to have a struct just for `change_info` and continue to return a tuple of `(transaction, change_info)` from `create_tx`

For example, here I introduce `CreateTxChange` and make its members match the names of the existing logic in fn `handle_commands`, does that make sense?

```
--- a/example-crates/example_cli/src/lib.rs
+++ b/example-crates/example_cli/src/lib.rs
@@ -181,21 +181,24 @@ impl core::fmt::Display for Keychain {
             Keychain::Internal => write!(f, "internal"),
         }
     }
 }
 
-// Alias changeset info in `create_tx`
-pub type ChangeSet = (keychain::ChangeSet<Keychain>, (Keychain, u32));
+pub struct CreateTxChange {
+    pub index_changeset: keychain::ChangeSet<Keychain>,
+    pub change_keychain: Keychain,
+    pub index: u32,
+}
 
 pub fn create_tx<A: Anchor, O: ChainOracle>(
     graph: &mut KeychainTxGraph<A>,
     chain: &O,
     keymap: &HashMap<DescriptorPublicKey, DescriptorSecretKey>,
     cs_algorithm: CoinSelectionAlgo,
     address: Address,
     value: u64,
-) -> anyhow::Result<(Transaction, Option<ChangeSet>)>
+) -> anyhow::Result<(Transaction, Option<CreateTxChange>)>
 where
     O::Error: std::error::Error + Send + Sync + 'static,
 {
     let mut changeset = keychain::ChangeSet::default();
 
@@ -385,11 +388,15 @@ where
             }
         }
     }
 
     let change_info = if selection_meta.drain_value.is_some() {
-        Some((changeset, (internal_keychain, change_index)))
+        Some(CreateTxChange {
+            index_changeset: changeset,
+            change_keychain: internal_keychain,
+            index: change_index,
+        })
     } else {
         None
     };
 
     Ok((transaction, change_info))
@@ -593,11 +600,16 @@ where
                 let graph = &mut *graph.lock().unwrap();
                 // take mutable ref to construct tx -- it is only open for a short time while building it.
                 let (tx, change_info) =
                     create_tx(graph, chain, keymap, coin_select, address, value)?;
 
-                if let Some((index_changeset, (change_keychain, index))) = change_info {
+                if let Some(CreateTxChange {
+                    index_changeset,
+                    change_keychain,
+                    index,
+                }) = change_info
+                {
                     // We must first persist to disk the fact that we've got a new address from the
                     // change keychain so future scans will find the tx we're about to broadcast.
                     // If we're unable to persist this, then we don't want to broadcast.
                     {
                         let db = &mut *db.lock().unwrap();
```

Reply:
Yes, sounds good!